### PR TITLE
feat(example): add an langchain-based chatbot example, and fix a bug

### PR DESCRIPTION
**Pull request by jianzs**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):

- Example
- TS Deducer
- Infra SDK

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

- enhance(sdk): remove the `fuzzyArn`, instead use the lazy value from pulumi.
  - Before, the type used to construct the permission list was `aws.types.input.iam.GetPolicyDocumentStatement` type, it only allows passing strings as arguments, and we used `fuzzyArn` to build the ARN through string concatenation. This approach is messy and unsafe. Now, we've switched the permission type to `aws.types.input.iam.GetPolicyDocumentStatementArgs`, which allows us to pass lazy values as arguments. This means we can configure the permission using the ARN from the Pulumi resource.
- fix(deducer): incorrectly generate duplicate client statements when one code section accesses a single resource multiple times
- feat(sdk): export  the table name and partition key of the created dynamodb instance to users
- feat(example): add an example, building a llama2 conversational chatbot with aws and langchain

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

### CODE REVIEW

File: examples/langchain-llama2-chatbot-sagemaker/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+- **Amazon DynamoDB**: Persisting session messages
+  - Table
+- **AWS Lambda**: Executing backend business logic
+  - Function, Role, Policy
+- **Amazon API Gateway**: Providing an HTTP API endpoint for user access
+  - Api, Route, Integration, Deployment, Stage
+
+## Comparison
+
+The table below shows a comparison between Pluto and other tools regarding the amount and type of code:
```
Comment (Peefy) [inline examples/langchain-llama2-chatbot-sagemaker/README.md:22]: Shouldn't these be compared to Lepton and its competitors for application developers? After all, Pluto's positioning is not actually an IaC tool.

File: examples/langchain-llama2-chatbot-sagemaker/README.md [hunk 2]
```diff
@@ -21,14 +21,21 @@ As illustrated above, our example application relies on the following AWS servic
...
+| [Pulumi](https://www.pulumi.com/)                       | Infrastructure Configuration       | 157(.ts)             | [Code](./comparison/pulumi)      |
+| [Pulumi Serverless](https://www.pulumi.com/serverless/) | Business Logic + Configuration     | 256(.ts)             | [Code](./comparison/pulumi-app)  |
+| [Winglang - TS](https://github.com/winglang/wing)       | Business Logic + Configuration     | 100(.ts)             | [Code](./comparison/wing-ts)     |
+| [Winglang - Wing](https://github.com/winglang/wing)     | Business Logic + Configuration     | 71(.w) + 47(.ts)     | [Code](./comparison/wing-wing)   |
+| [Lepton](https://www.lepton.ai/)                        | Business Logic + Configuration     | 116(.py)             | [Code](./comparison/lepton)      |
+| [Serverless](https://github.com/serverless/serverless)  | Business Logic + Configuration     | 79(.ts) + 120(.yaml) | [Code](./comparison/serverless)  |
+| **Pluto**                                               | **Business Logic + Configuration** | **100(.ts)**         | [Code](./src)                    |
```
Comment (Peefy) [inline examples/langchain-llama2-chatbot-sagemaker/README.md:34]: How many lines is the Python code approximately?

Comment (jianzs) [reply examples/langchain-llama2-chatbot-sagemaker/README.md:34]: If it's Pluto, around 72 lines, not sure about other tools.

Comment (Peefy) [reply examples/langchain-llama2-chatbot-sagemaker/README.md:34]: > If it's Pluto, around 72 lines, not sure about other tools.

You can write these information in this document.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**